### PR TITLE
3285/add functionality for displaying multiple token serials in the audit log

### LINKED ISSUE #3285: Display of multiple token serials in the audit log
It would be nice if multiple token serials could also be displayed in the audit log.

### PR DESCRIPTION
**Pull request by NNLemling**
closes #3285

### CODE REVIEW

File: privacyidea/api/auth.py [hunk 1]
```diff
@@ -309,6 +309,8 @@ def get_auth_token():
                                                     superuser_realms=
                                                     superuser_realms)
         details = details or {}
+        serials = ",".join([challenge_info["serial"] for challenge_info in details["multi_challenge"]]) \
```
Comment (plettich) [inline privacyidea/api/auth.py:312]: If You use `', '.join()` You'll have a space after the comma, looks better.

File: privacyidea/api/validate.py [hunk 2]
```diff
@@ -416,10 +416,11 @@ def check():
                     # additional attributes
                     for k, v in ui.items():
                         result["attributes"][k] = v
-
+    serials = ",".join([challenge_info["serial"] for challenge_info in details["multi_challenge"]]) \
```
Comment (plettich) [inline privacyidea/api/validate.py:419]: same as above

File: privacyidea/api/validate.py [hunk 3]
```diff
@@ -416,10 +416,11 @@ def check():
...
+        if 'multi_challenge' in details else details.get('serial')
     g.audit_object.log({"info": log_used_user(user, details.get("message")),
                         "success": success,
-                        "serial": serial or details.get("serial"),
+                        "serial": serials,
```
Comment (plettich) [inline privacyidea/api/validate.py:423]: We should also add a testcase for this and update the frontend code to create the links

**Review by plettich — CHANGES_REQUESTED** [review]

File: privacyidea/api/auth.py (hunk 1 above)
Comment (NNLemling) [reply privacyidea/api/auth.py:312]: If a space is added after the "," the link to the token contains a space that no longer works.

File: privacyidea/api/validate.py (hunk 2 above)
Comment (NNLemling) [reply privacyidea/api/validate.py:419]: explanation above

File: privacyidea/api/validate.py (hunk 3 above)
Comment (NNLemling) [reply privacyidea/api/validate.py:423]: The frontend code actually handles the serials correctly. Links to the various tokens are generated and work.
serials2

**Review by plettich — APPROVED** [review]

**Pull request merged**